File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/StringSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.exceptions.ValidationException;

import java.util.List;
import java.util.Set;

public interface StringSchemaValidator {
    String getNewInstance(String arg, List<Object> pathToItem, PathToSchemasMap pathToSchemas);
    String validate(String arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/StringSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface StringSchemaValidator<T> {
    @Throws(ValidationException::class)
    fun validate(arg: String, configuration: SchemaConfiguration?): String

    @Throws(ValidationException::class)
    fun validateAndBox(arg: String, configuration: SchemaConfiguration?): T
}